pilot/pkg/config/coredatamodel/discovery.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coredatamodel
+
+import (
+	"errors"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+
+	networking "istio.io/api/networking/v1alpha3"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/config"
+	"istio.io/pkg/log"
+)
+
+var notReadyEndpointkey = "networking.alpha.istio.io/notReadyEndpoints"
+
+// Discovery implements all the necessary interfaces
+// to satisfy the aggregated controller, it is used to
+// expose notReadyEndPoints that are received via MCP
+type Discovery interface {
```

Comment:
Do you need this interface? Or is it sufficient that `MCPDiscovery` (which you could then rename to just `Discovery`) below fulfills these methods?

You can ensure that by doing something like this:

```go
var _ model.Controller = &MCPDiscovery{}
ver _ model.ServiceDiscovery = &MCPDiscovery{}
```

This guarantees that MCPDiscovery "is-a" Controller and ServiceDiscovery.

Reply:
No need for an interface, I will update.